pkg/cri/resource-manager/cache/pod.go

```diff
@@ -372,43 +372,47 @@ func (p *pod) GetQOSClass() v1.PodQOSClass {
 }
 
 // GetContainerAffinity returns the annotated affinity for the named container.
-func (p *pod) GetContainerAffinity(name string) []*Affinity {
+func (p *pod) GetContainerAffinity(name string) ([]*Affinity, error) {
 	if p.Affinity != nil {
-		return (*p.Affinity)[name]
+		return (*p.Affinity)[name], nil
 	}
 
-	p.Affinity = &podContainerAffinity{}
+	affinity := &podContainerAffinity{}
```

Comment:
No, the idea is that the function keeps failing. That's why I had to change it to only set the parsed pod annotations on success as that is the easiest way to achieve this with the current code. An alternative would have been to always pre-parse all affinity annotations on the pod when it is created and reject the pod already. I just was not comfortable doing such a larger change now. BTW, the only reason why I noticed this because it looks like recent kubelets don't take no for an answer immediately but rather retry a few times in case of an error. Without such a change then, the first attempt failed but already the second one succeeded, incorrectly without getting any assigned affinity.